moped-api/users/users.py

```diff
@@ -218,81 +217,49 @@ def user_update_user(id: str, claims: list) -> (Response, int):
         )
 
         if not profile_valid:
             return jsonify({"error": profile_error_feedback}), 400
 
-        # The status is active, check if the user exists
-        if is_deleted == False:
-            try:
-                cognito_client.admin_get_user(UserPoolId=USER_POOL, Username=id)
-                user_already_exists = True
-            except ClientError as e:
-                user_already_exists = False
-        # If the user doesn't exist and is now active, create it...
-        if user_already_exists == False and is_deleted == False:
-            # Check if the password needs to be provided
-            if password is None or password == "":
-                return jsonify({"error": {"message": "No password provided."}}), 400
-
-            # If we have all we need, then we proceed
-            try:
-                cognito_response = cognito_client.admin_create_user(
-                    UserPoolId=USER_POOL,
-                    Username=email,
-                    TemporaryPassword=password,
-                    UserAttributes=[
-                        {"Name": "email", "Value": email},
-                        {"Name": "email_verified", "Value": "true"},
-                    ],
-                )
-                # Then  we must set the user password
-                cognito_username = cognito_response["User"]["Username"]
-                cognito_client.admin_set_user_password(
-                    UserPoolId=USER_POOL,
-                    Username=cognito_username,
-                    Password=password,
-                    Permanent=True,
-                )
-                # Copy the username to the UUID variable
-                user_cognito_uuid = cognito_username
-                id = cognito_username
-                reactivate_account = True
-            except ClientError as e:
-                return jsonify(e.response), 400  # Bad request
+        # If a password was included, let's reset it
+        if password is not None and password != "":
+            cognito_client.admin_set_user_password(
+                UserPoolId=USER_POOL,
+                Username=cognito_user_id,
+                Password=password,
+                Permanent=True,
+            )
 
-        # Retrieve current profile (to fetch old email)
-        user_info = cognito_client.admin_get_user(UserPoolId=USER_POOL, Username=id)
+        # Retrieve current profile to fetch old email in case a new one was in the payload
+        user_info = cognito_client.admin_get_user(
+            UserPoolId=USER_POOL, Username=cognito_user_id
+        )
         user_email_before_update = get_user_email_from_attr(user_attr=user_info)
 
-        json_data = request.json
-        roles = json_data.get("roles", None)
-
-        user_profile = generate_user_profile(cognito_id=id, json_data=request.json)
-
-        db_response = db_update_user(
-            user_profile=user_profile, search_by_email=reactivate_account
```

Comment:
This `search_by_email` arg was meant to handle the case of user activation. Normally, during a user update, we search by Cognito user ID, but, in the case of activation, we don't have that detail anymore so we would have to use the user email to find their User table row.

This is no longer needed since this route no longer handles user activation.